internal/internal_workflow.go

```diff
@@ -584,6 +585,17 @@ func (c *channelImpl) receiveAsyncImpl(callback *receiveCallback) (v interface{}
 		r := c.buffer[0]
 		c.buffer[0] = nil
 		c.buffer = c.buffer[1:]
+
+		// Move blocked sends into buffer
+		if len(c.blockedSends) > 0 {
+			b := c.blockedSends[0]
+			c.blockedSends[0] = nil
+			c.blockedSends = c.blockedSends[1:]
+			if b.fn() {
```

Comment:
I think it should iterate through blockedSends until b.fn() returns true. In this case if it returns false nothing is added to the buffer.

Reply:
I think you have to use Select that matches a different branch than send to reproduce this edge case.

Reply:
Updated and created a test case for it.